fda/grid_interpolation.py

```diff
@@ -237,203 +233,48 @@ def _spline_evaluator_n_m(spl, t, derivative):
 
         return spline
 
-    #def evaluate(self, t, derivative=0, grid=False):
-    def evaluate(self, eval_points, derivative=0, extrapolation=None,
-                 grid=False, keepdims=None):
-
-        derivative = self._process_derivative(derivative)
-
-        # Case evaluation of a numeric value
-        if isinstance(eval_points, numbers.Number):
-            eval_points = [eval_points]
-
-        eval_points = numpy.array(eval_points, dtype=numpy.float)
-
-        if grid:
-            return self.evaluate_grid(eval_points, derivative,
-                                      extrapolation=extrapolation,
-                                      keepdims=keepdims)
-
-        if ((self._ndim_domain == 1 and len(eval_points.shape) == 2) or
-            len(eval_points.shape) == 3):
-
-            return self.evaluate_composed(eval_points, derivative,
-                                          extrapolation=extrapolation,
-                                          keepdims=keepdims)
-
-        return self.evaluate_spline(eval_points, derivative,
-                                    extrapolation=extrapolation,
-                                    keepdims=keepdims)
-
-    def evaluate_spline(self, t, derivative=0, extrapolation=None, keepdims=None):
 
-        if extrapolation is not None:
-            # Coordinates with shape (n_evalpoints x ndim_image)
-            eval_points_coord = t.reshape((len(t), self._ndim_domain))
+    def evaluate(self, eval_points, *, derivative=0):
 
+        derivative = self._process_derivative(derivative)
 
-            # Boolean index of points where the extrapolation should be applied
-            index_ext = _extrapolation_index(eval_points_coord, self._domain_range)
-
-            # Flag to apply extrapolation
-            extrapolate = index_ext.any()
-
-        else:
-            extrapolate = False
-
-
-        if extrapolate: # Index of points with extrapolation
-            index = ~ index_ext
-            t_eval = t[index]
-
-        else: # Else all points will be evaluated
-            t_eval = t
 
         # Constructs the evaluator for t_eval
         if self._ndim_image == 1:
-            evaluator = lambda spl: self._spline_evaluator(spl[0], t_eval,
+            evaluator = lambda spl: self._spline_evaluator(spl[0], eval_points,
```

Comment:
Lambdas are for anonymous functions ONLY. Use `def` instead if you want to name them.